drivers/flash/flash_stm32f7x.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <string.h>
+#include <flash.h>
+#include <init.h>
+#include <soc.h>
+
+#include "flash_stm32.h"
+
+#define STM32F7X_SECTOR_MASK		((u32_t) 0xFFFFFF07)
+
+bool flash_stm32_valid_range(struct device *dev, off_t offset, u32_t len,
+			     bool write)
+{
+	ARG_UNUSED(write);
+
+	return flash_stm32_range_exists(dev, offset, len);
+}
+
+static int write_byte(struct device *dev, off_t offset, u8_t val)
+{
+	struct stm32f7x_flash *regs = FLASH_STM32_REGS(dev);
+	u32_t tmp;
+	int rc;
+
+	/* if the control register is locked, do not fail silently */
+	if (regs->cr & FLASH_CR_LOCK) {
+		return -EIO;
+	}
+
+	rc = flash_stm32_wait_flash_idle(dev);
+	if (rc < 0) {
+		return rc;
+	}
+
+	regs->cr &= ~CR_PSIZE_MASK;
+	regs->cr |= FLASH_PSIZE_BYTE;
+	regs->cr |= FLASH_CR_PG;
+
+	/* flush the register write */
+	tmp = regs->cr;
```

Comment:
The reference manual recommends to use a DSB instruction to flush the register. The Cortex-M7 is using a more advanced pipeline, so the trick of doing a dummy read might not be enough.